src/main/webapp/app/exercises/shared/submission/submission.service.ts

```diff
@@ -167,14 +167,22 @@ export class SubmissionService {
         return res.clone({ body });
     }
 
+    public convertSubmissionResponseFromServer<T extends Submission>(response: HttpResponse<T>): HttpResponse<T> {
+        const submission = response.body!;
+        setLatestSubmissionResult(submission, getLatestSubmissionResult(submission));
+        this.setSubmissionAccessRights(submission);
+        SubmissionService.convertConnectedParticipationFromServer(submission);
+        return response;
```

Comment:
This looks duplicated, I think you can also call the the method below:
```suggestion
        return convertSubmissionFromServer(response);
```